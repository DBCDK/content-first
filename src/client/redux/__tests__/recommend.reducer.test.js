import recommendReducer, {
  RECOMMEND_REQUEST,
  RECOMMEND_RESPONSE
} from '../recommend.reducer';

describe('recommendReducer', () => {
  test('normalized key is used for request, tags only', () => {
    const state = recommendReducer(undefined, {
      type: RECOMMEND_REQUEST,
      tags: [2, 1, 4]
    });
    expect(state).toMatchSnapshot();
  });
  test('normalized key is used for request, creators only', () => {
    const state = recommendReducer(undefined, {
      type: RECOMMEND_REQUEST,
      creators: ['ib', 'arne']
    });
    expect(state).toMatchSnapshot();
  });
  test('normalized key is used for request, tags and creators', () => {
    const state = recommendReducer(undefined, {
      type: RECOMMEND_REQUEST,
      tags: [2, 1, 4],
      creators: ['ib', 'arne']
    });
    expect(state).toMatchSnapshot();
  });
  test('normalized key is used for response', () => {
    const state = recommendReducer(undefined, {
      type: RECOMMEND_RESPONSE,
      tags: [2, 1, 4],
      creators: ['ib', 'arne'],
      pids: ['pid1', 'pid2', 'pid3']
    });
    expect(state).toMatchSnapshot();
  });
});
