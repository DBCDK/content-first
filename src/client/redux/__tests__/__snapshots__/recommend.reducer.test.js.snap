// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommendReducer normalized key is used for request, creators only 1`] = `
Object {
  "recommendations": Object {
    "arne-ib": Object {
      "isLoading": true,
    },
  },
}
`;

exports[`recommendReducer normalized key is used for request, tags and creators 1`] = `
Object {
  "recommendations": Object {
    "1-2-4arne-ib": Object {
      "isLoading": true,
    },
  },
}
`;

exports[`recommendReducer normalized key is used for request, tags only 1`] = `
Object {
  "recommendations": Object {
    "1-2-4": Object {
      "isLoading": true,
    },
  },
}
`;

exports[`recommendReducer normalized key is used for response 1`] = `
Object {
  "recommendations": Object {
    "1-2-4arne-ib": Object {
      "error": undefined,
      "isLoading": false,
      "pids": Array [
        "pid1",
        "pid2",
        "pid3",
      ],
    },
  },
}
`;
